tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/ExecuteAsyncIntegrationExtra.cs

```diff
@@ -40,22 +38,10 @@ internal static CallTargetState OnMethodBegin<TTarget, TOperationContext>(TTarge
         {
             var operation = operationContext.Operation;
             var operationType = HotChocolateCommon.GetOperation(operation.OperationType);
-            var operationName = GetNameStringValue(operation.Name.Value);
+            var operationName = operation.Name?.Value;
 
             HotChocolateCommon.UpdateScopeFromExecuteAsync(Tracer.Instance, operationType, operationName);
             return CallTargetState.GetDefault();
         }
-
-        private static string GetNameStringValue(object nameValue)
-        {
-            // Get the string value of the NameString
-            // The NameString value can be either another NameString or a string
```

Comment:
the property is defined here: https://github.com/ChilliCream/graphql-platform/blob/c162ca29c23acc69cf81a33155d7384ad4dc9d0f/src/HotChocolate/Core/src/Types/Resolvers/IOperation.cs#L30
It's a struct that looks like a nullable, which is itself a nullable, so there is 2 levels of `.Value` to get, which explains the initial confusion around this I think.

Reply:
how can the `.Value` be null when they are actual non-nullable struct ?
This is not the latest code that we are commenting on too, it now looks like `operation.Name.HasValue ? operation.Name.Value.Value : null`
`operation` is not null/default because we check `HasValue` before, so here we check name, and then name.Value, any of those can be null.
In any case, we do update the span even if one of those values is null.

Reply:
Are we talking about this `NameString`?

https://github.com/ChilliCream/graphql-platform/blob/12.22.6/src/HotChocolate/Core/src/Abstractions/NameString.cs

If so, this is a custom `struct`, not a `Nullable<T>`. (`Nullable<string>` is not valid, anyway.)

Reply:
Just trying to understand the issue :)

Reply:
Yes that's right, it's a nullable of a custom struct, but we need a proxy both for the nullable<NameString> and the NameString struct